library/graylog_streams.py

```diff
@@ -381,27 +381,88 @@ def create(module, base_url, headers, index_set_id):
 
     return info['status'], info['msg'], content, url
 
-
-def create_rule(module, base_url, headers):
+def query_rules(module, base_url, headers):
 
     url = "/".join([base_url, module.params['stream_id'], "rules"])
-
+    #raise Exception(url)
     payload = {}
+    field = module.params['field']
+    value = module.params['value']
+    #raise Exception(module.params['field'])
 
-    for key in ['field', 'type', 'value', 'inverted', 'description']:
-        if module.params[key] is not None:
-            payload[key] = module.params[key]
-
-    response, info = fetch_url(module=module, url=url, headers=json.loads(headers), method='POST', data=module.jsonify(payload))
 
-    if info['status'] != 201:
+    #TODO: GET current rules, compare field and value to determine duplication
+    response, info = fetch_url(module=module, url=url, headers=json.loads(headers), method='GET')
+    if info['status'] != 200:
         module.fail_json(msg="Fail: %s" % ("Status: " + str(info['msg']) + ", Message: " + str(info['body'])))
 
     try:
         content = to_text(response.read(), errors='surrogate_or_strict')
+        rules = json.loads(content)
+        #raise Exception(rules)
     except AttributeError:
         content = info.pop('body', '')
 
+    if rules is not None:
+        i = 0
+        while i < len(rules['stream_rules']):
```

Comment:
This is better written as `for i in range(len(rules['stream_rules']):` in Python. (and remove the `i = 0` and `i += 1` of course)

Reply:
Much cleaner indeed.

Reply:
Oh, wait, you can go one (important step) further. Sorry, missed this the first time:
```
for rule in rules['stream_rules']:
```
and then remove line 409

Reply:
Moral of the story, if your Python code ends up looking like C/C++, then you prob. need to re-work it. :laughing:

Reply:
Jinx! We arrived at the same loop iteration snippet independently which makes me feel pretty good :)